### PR TITLE
Add openDialog checks

### LINKED ISSUE #126: Rule: %s_remote_uri
Implement the `%s_remote_uri` rule for JS validator.

From https://github.com/mozilla/addons-validator/blob/master/docs/rules.md#call-definitions

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #126 

Made the files and rule names from eslints PoV all lowercase rather than a mix of lower + camel + underscores. This matches closer to typical eslint naming.

* add tests for non-literal uri arg to openDialog.
* add checks for remote uri passed to openDialog.
* It kills off all the regex - negating matching data: but allowing other protocols started to make the regexes too gnarly. Instead I've replaced the regex with simpler utils that actually parse the urls. 
* css uris are now extracted using via regex in utils. 

TODO: this doesn't do anything to check whether CSS uri values are escaped correctly e.g. ' " and () in un-quoted CSS uris [1] and also we could use something to check css quotes match etc too. I'll file another issue for that.

[1] https://drafts.csswg.org/css-values-3/#urls

### CODE REVIEW

**Comment by tofumatt** [comment]
Looks like the existing JS rules weren’t moved so would still have weird filenames.

Getting rid of those weird names is several 👍 from me!

**Comment by muffinresearch** [comment]
Case preserving filesystems FTL. :/

File: src/rules/css/detectBadMozBinding.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import * as messages from 'messages';
```
Comment (tofumatt) [inline src/rules/css/detectBadMozBinding.js]: I like that our JS files now aren't `camelCase` or whatever that's called, can we do the same for CSS?

File: src/utils.js [hunk 2]
```diff
@@ -88,3 +90,36 @@ export function ignorePrivateFunctions(list) {
 
   return filteredList;
 }
+
+
+export function isLocalUrl(url_) {
```
Comment (tofumatt) [inline src/utils.js]: `url_` is a weird variable name, did you mean `_url`?

File: tests/rules/javascript/test.opendialog_remote_uri.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                foo.openDialog("ftps://foo.com/bar/");
+                foo.openDialog("ftp://foo.com/bar/");
+                foo.openDialog("//foo.com/bar/");
+                foo.openDialog("data:whatever");`;
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    return jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 6);
+        for (let msg of validationMessages) {
```
Comment (tofumatt) [inline tests/rules/javascript/test.opendialog_remote_uri.js]: Nitpick here, but I always like full variable names when we can get them, so `message` over `msg` if it's possible.

**Comment by tofumatt** [comment]
I like that functions over wild regexs and the prettier filenames :+1: 

r+wc, if the CSS filenames can be made more like `detect_bad_moz_binding` that'd be great and there are a few variable name nitpicks.

File: src/utils.js (hunk 2 above)
Comment (muffinresearch) [inline src/utils.js]: Hmm, I should probably use es6 modules import syntax to modify the url lib or just import parse and then I can use url here.

File: tests/rules/javascript/test.opendialog_remote_uri.js (hunk 3 above)
Comment (muffinresearch) [inline tests/rules/javascript/test.opendialog_remote_uri.js]: Yeah fair enough. I tend to be little bit less worried about that for tests. But I suppose a consistent rule here is a bit more important than anything else.

**Comment by muffinresearch** [comment]
I'll file a bug for rule file name changes elsewhere.

**Pull request merged**